extension-workspace-reference.md

```diff
@@ -0,0 +1,121 @@
+# workspace/xreference extension to LSP
```

Comment:
I actually responded to your original comment about this a few days ago [here](https://github.com/sourcegraph/language-server-protocol/pull/8#discussion_r87575314). To quote myself:

> I would agree with you (in fact, that was the original name), but workspace/symbol isn't workspace/symbols.. so I think it should be named this way. Let me know if you can think of a compelling reason otherwise and I'll change it!